src/experiments.ts

```diff
@@ -0,0 +1,182 @@
+  experiment("rtdbmanagement", "TODO"),
+
+  // Extensions experiments
+  experiment("ext", "Enables the 'ext:sources:create' command"),
+  experiment("extdev", "Enalbes the 'ext:dev' family of commands"),
+
+  // Cloud Functions for Firebase experiments
+  experiment("pythonfunctions", "Python support for Cloud Functions for Firebase", {
+    fullDescription:
+      "Adds the ability to initializea and deploy Cloud " +
+      "Functions for Firebase in Python. While this feature is experimental " +
+      "breaking API changes are allowed in MINOR API revisions",
+  }),
+  experiment("golang", "Golang support for Cloud Functions for Firebase. Does not work.", {
+    fullDescription:
+      "Code paths to play with adding support for Golang to Cloud " +
+      "Functions for Firebase. This was written to target 1st gen functions, which " +
+      "will never be released (all new languages will be 2nd gen only). If you " +
+      "are lurking in our experiments folder and are excited about the prospect of " +
+      "Golang support, tell our support team.",
+  }),
+  experiment(
+    "deletegcfartifacts",
+    `Add the ${clc.bold(
+      "firebase functions:deletegcfartifacts"
+    )} command to purge docker build images`,
+    {
+      fullDescription:
+        `Add the ${clc.bold("firebase functions:deletegcfartifacts")}` +
+        "command. Google Cloud Functions creates Docker images when building your " +
+        "functions. Cloud Functions for Firebase automatically cleans up these " +
+        "images for you on deploy. Customers who predated this cleanup, or customers " +
+        "who also deploy Google Cloud Functions with non-Firebase tooling may have " +
+        "old Docker images stored in either Google Container Repository or Artifact " +
+        `Registry. The ${clc.bold("firebase functions:deletegcfartifacts")} command ` +
+        "will delete all Docker images created by Google Cloud Functions irrespective " +
+        "of how that image was created.",
+      public: true,
+    }
+  ),
+  experiment("functionsparams", "Adds support for paramaterizing functions deployments"),
+  experiment(
+    "skipdeployingnoopfunctions",
+    "Detect that there have been no changes to a function and skip deployment"
+  ),
+
+  // Emulator experiments
+  experiment("emulatoruisnapshot", "Load pre-release versions of hte emulator UI"),
+
+  // Hosting experiments
+  experiment("frameworkawareness", "Native support for popular web frameworks", {
+    fullDescription:
+      "Adds support for popular web frameworks such as Next.js " +
+      "Nuxt, Netlify, Angular, and Vite-compatible frameworks. Firebase is " +
+      "committed to support these platforms long-term, but a manual migration " +
+      "may be required when the non-experimental support for these frameworks " +
+      "is released",
+  }),
+
+  // Access experiments
+  experiment("crossservicerules", "Allow Firebase Rules to reference resources in other services"),
```

Comment:
@sam-gc Looks like you added this preview. Can you sign off or reword the short description? Should we make this experiment visible when customers call `firebase experiments:list`? If so, can you also provide a full description and/or docs?

Can you also verify/correct my usage of this experiment? Later in this PR I assert that the experiment is enabled, which breaks the test. In the "experiments" world, if we see a clear intent to use an experiment (e.g. if there are rules with cross service dependencies) we should assert that the experiment is enabled rather than fail silently. I turned this into an assert, but that causes a unit test to fail and I want to defer to you for the correct behavior.

Reply:
Will Not Fix (Obsolete). Flag was removed in https://github.com/firebase/firebase-tools/pull/5002